eng/pipelines/coreclr/templates/crossgen2-comparison-build-job.yml

```diff
@@ -0,0 +1,169 @@
+    - name: artifactsDirectory
+      value: $(Build.SourcesDirectory)$(dir)artifacts
+    - name: binDirectory
+      value: $(artifactsDirectory)$(dir)bin
+    - name: productDirectory
+      value: $(binDirectory)$(dir)coreclr
+    - name: workItemDirectory
+      value: $(artifactsDirectory)$(dir)tests$(dir)coreclr$(dir)$(targetFlavor)$(dir)Tests$(dir)CrossCompileRoot
+    - name: crossgencompare_build_artifact
+      value: crossgen_comparison_build_$(osGroup)$(osSubgroup)_$(archType)
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - name: target_crossgen2_os
+        value: windows
+    - ${{ if eq(parameters.osGroup, 'Linux') }}:
+      - name: target_crossgen2_os
+        value: linux
+    - ${{ if eq(parameters.osGroup, 'OSX') }}:
+      - name: target_crossgen2_os
+        value: osx
+    - name: crossgen2location
+      value: $(productDirectory)$(dir)$(targetFlavor)$(dir)crossgen2$(dir)crossgen2.dll
+    - ${{ if ne(parameters.archType, 'x64') }}:
+      - name: crossgen2location
+        value: $(productDirectory)$(dir)$(targetFlavor)$(dir)x64$(dir)crossgen2$(dir)crossgen2.dll
+    - name: librariesProductDllDir
+      value: $(Build.SourcesDirectory)$(dir)artifacts$(dir)bin$(dir)runtime$(dir)net5.0-$(osGroup)$(osSubgroup)-$(buildConfig)-$(archType)
+
+    - ${{ parameters.variables }}
+
+    # Test job depends on the corresponding build job
+    dependsOn:
+    - ${{ format('coreclr_{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
+    - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
+      - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
+
+    # Run all steps in the container.
+    # Note that the containers are defined in platform-matrix.yml
+    container: ${{ parameters.container }}
+    timeoutInMinutes: 180 # 3 hrs
+
+    steps:
+
+    # Download product build
+    - template: /eng/pipelines/common/download-artifact-step.yml
+      parameters:
+        unpackFolder: $(buildProductRootFolderPath)
+        artifactFileName: '$(buildProductArtifactName)$(archiveExtension)'
+        artifactName: '$(buildProductArtifactName)'
+        displayName: 'product build'
+
+    # Optionally download live-built libraries
+    - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
+      - template: /eng/pipelines/common/download-artifact-step.yml
+        parameters:
+          unpackFolder: $(librariesDownloadDir)
+          cleanUnpackFolder: false
+          artifactFileName: '$(librariesBuildArtifactName)$(archiveExtension)'
+          artifactName: '$(librariesBuildArtifactName)'
+          displayName: 'live-built libraries'
+
+    # Populate Core_Root
+    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) $(buildConfig) $(archType) $(crossArg) generatelayoutonly
```

Comment:
I guess that at some point we should expand this testing to exercise CG2-compiled framework (i.e. <code>crossgen2</code> or <code>composite</code> extra command-line parameters) but that can easily wait for a follow-up change.

Reply:
I don't think we really need that. In fact, looking closer, I don't actually think we need to populate core_root in this job as I believe we don't use it.

Its more interesting to think about the crossgen2-comparison-job.yml where we do use the generated runtime to run crossgen2 itself. In that case there may be a meaningful benefit to running crossgen2 when producing the core_root. OTOH, that may just make it harder to reproduce issues found by these tests.